src/app/cpe/project/project.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import {NavigationService} from "../../routing-module/navigation.service";
+import {Project} from "../../repository/repository-model";
+import {ProjectRepositoryService} from "../../repository/project-repository.service";
+import {Subscription} from "rxjs";
+import {ActivatedRoute} from "@angular/router";
+
+
+@Component({
+  selector: 'app-project',
+  templateUrl: './project.component.html'
+})
+export class ProjectComponent implements OnInit, OnDestroy {
+
+  private _projectId = '';
+  private _project: Project | undefined = undefined;
+
+  private projectRouteSubscription: Subscription;
+
+  constructor( private projectService: ProjectRepositoryService,
+               private route: ActivatedRoute,
+               private naviService: NavigationService ) {
+    this.projectRouteSubscription = new Subscription();
+  }
+
+  ngOnInit(): void {
+    this.projectRouteSubscription = this.route.params.subscribe(parameter => {
+      if (parameter['id'] && parameter['id'] !== this._projectId) {
+        this.initProject(parameter['id']);
+      }
+    });
+  }
+
+  get project() {
+    return this._project;
+  }
+
+  /**
+   * inits a specific project id passed. If there is no project with that id, the app navigates
+   * to the projects component
+   * @param projectId: The project which is loaded.
+   */
+  initProject(projectId: string | null) {
+    if ( projectId ) {
+      this.loadProject(projectId);
+    } else { // if no project id given, route to default project page
+      console.warn(`No project id passed - navigating to projects overview.`)
+      this.goToProjectsOverview();
+    }
+  }
+
+  /**
+   * Loads the project's data. If there is no data for the given project id, it navigates to the projects component.
+   * @param projectId: The project to which is navigated.
+   */
+  loadProject(projectId: string) {
+    this._projectId = projectId;
+    this.projectService.getProjectByID(this._projectId).then(p => {
+      if (p) {
```

Comment:
You are checking p if it could be undefined or null. That is bad code. getProjectByID() should not return undefined or null

Reply:
This is a **fake api** with **mock components** for having any **routing** ready and develop a data driven **prove of concept**. We want to use https://github.com/Effect-TS/effect  anyways - so no fetch, no htttpclient get at all. There will be a NullProject of course. This will be removed anyways!  No need to blame a fake api with fake components. C'mon please ... That's like saying your services shouldn't be hardcoded when you mock them.